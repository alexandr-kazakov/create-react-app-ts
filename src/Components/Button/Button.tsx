export type ButtonProps = {
    text: string;
    onClick?: () => void;
    className?: string;
    testId?: string;
}

const Button: React.FC<ButtonProps> = ({text, onClick, className, testId}) => {
    return (
       <button onClick={onClick} className={className} data-testid={testId}>{text}</button>
    )
}

export default Button;