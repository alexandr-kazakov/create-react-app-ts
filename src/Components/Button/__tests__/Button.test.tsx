import { render, fireEvent, screen } from '@testing-library/react';
import Button, { ButtonProps } from '../Button';

describe('Button component', () => {
  const defaultProps: ButtonProps = {
    text: 'Click me',
  };

  test('renders button with text', () => {
    render(<Button {...defaultProps} />);
    expect(screen.getByText('Click me')).toBeInTheDocument();
  });

  test('calls onClick when clicked', () => {
    const onClickMock = jest.fn();
    render(<Button {...defaultProps} onClick={onClickMock} />);
    fireEvent.click(screen.getByText('Click me'));
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  test('applies className to button', () => {
    render(<Button {...defaultProps} className="custom-class" />);
    const button = screen.getByRole('button');
    expect(button).toHaveClass('custom-class');
  });
});

describe('Button component snapshot', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(<Button text="Click me" />);
    expect(asFragment()).toMatchSnapshot();
  });
});