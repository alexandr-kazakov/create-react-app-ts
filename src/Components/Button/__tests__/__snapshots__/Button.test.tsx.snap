// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot matches snapshot 1`] = `
<DocumentFragment>
  <button>
    Click me
  </button>
</DocumentFragment>
`;
